v0/destinations/google_ads/transform.js

```diff
@@ -0,0 +1,169 @@
+      if (
+        get(message, "properties.affiliation") === "App Store" ||
+        get(message, "properties.affiliation") === "Google Store"
+      ) {
+        return "in_app_purchase";
+      }
+      return "ecommerce_purchase";
+    case "product list viewed":
+      return "view_item_list";
+    case "product added":
+      return "add_to_cart";
+    case "poducts searched":
+      return "view_search_results";
+    case "product viewed":
+      return "view_item";
+    default:
+      return "custom";
+  }
+}
+function responseBuilderSimple(message, category, destination) {
+  const { Config } = destination;
+  const { devToken, linkId, version, lat, fallbackIDFAZero } = Config;
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (payload) {
+    payload.dev_token = devToken;
+    payload.link_id = linkId;
+    payload.app_event_type = setEventType(message);
+    // This field is required if app_event_type is custom.
+    if (payload.app_event_type === "custom") {
+      payload.app_event_name = setEventName(message);
+    }
+    payload.lat = lat ? 1 : 0;
+    /**
+     * The type of identifier stored in the rdid field
+     * For Android: advertisingid
+     * For iOS: idfa
+     */
+    if (get(message, "context.device.type") === "Android") {
+      payload.id_type = "advertisingid";
+    } else {
+      payload.id_type = "idfa";
+    }
+    // if fall back to Idfa zero setting is true and no rdid is not present it will be set to default value
+    if (fallbackIDFAZero && !payload.rdid) {
+      payload.rdid = "00000000-0000-0000-0000-000000000000";
+    }
+    // if fall back to value is false and rdid is not present the message will not be accepted by google ads so it will be aborted by rudder.
+    else if (!fallbackIDFAZero && !payload.rdid) {
+      throw new Error("No advertisingId set. Aborting message");
+    }
+    const response = defaultRequestConfig(message);
+    response.endpoint = `${baseEndpoint}/${version}`;
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = get(message, "properties");
+    response.params = payload;
+    /**
+     * if the request body is empty (in cases where no rich event data is passed in the app_event_data payload),
+     * Google ads server requires that we explicitly set the Content-Length: 0 header on your request.
+     */
+    if (response.body.JSON) {
+      response.headers = {
+        "User-Agent": get(message, "context.userAgent"),
+        "X-Forwarded-For":
+          get(message, "reques_ip") || get(message, "context.ip"),
+        "Content-Type": "application/json; charset=utf-8"
+      };
+    } else {
+      response.headers = {
+        "User-Agent": get(message, "context.userAgent"),
+        "X-Forwarded-For":
+          get(message, "reques_ip") || get(message, "context.ip"),
+        "Content-Type": "application/json; charset=utf-8",
+        "Content-Length": 0
+      };
+    }
```

Comment:
Can we please optimize the code here without repetition?

Also, for  `IP` we should give precedence to `context.ip` over `request_ip`

Reply:
done